hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/Parameters.java

```diff
@@ -13,70 +16,144 @@
  */
 public class Parameters {
 
-	private boolean inString;
-	private boolean inQuoted;
-	private boolean inSqlComment;
-	private boolean inCComment;
-	private boolean escaped;
-	private int count = 0;
-	private StringBuilder result;
-	private int previous;
-
-	private Parameters(String sql, int parameterCount) {
-		result = new StringBuilder( sql.length() + parameterCount );
-		sql.codePoints().forEach(this::append);
+	private static Parameters INSTANCE = new Parameters();
+
+	private static Parameters NO_PARSING = new Parameters() {
+		@Override
+		public String process(String sql) {
+			return sql;
+		}
+
+		@Override
+		public String process(String sql, int parameterCount) {
+			return sql;
+		}
+
+		@Override
+		public String processLimit(String sql, Object[] parameterArray, boolean hasOffset) {
+			return sql;
+		}
+	};
+
+	public static Parameters create(Dialect dialect) {
```

Comment:
Could we rename this to, I dunno, `for()`, or `get()`, or `instance()`? Because it doesn't actually instantiate anything.

Reply:
Good point, I will change it to `for`